### ISSUE TITLE
[BUG] fixConfigurableProductQty critical error

### ISSUE THREAD

**Issue by rjm87** [tig-nl/postnl-magento2#278]
Hi, we are running Magento 2.4.3-p1 and are experiencing the following error when using postnl-magento2 v1.10.0. It prevents users to place an order on the checkout page.

`
[2021-11-09 13:48:44] main.CRITICAL: Notice: Trying to access array offset on value of type null in /data/web/magento2/vendor/tig/postnl-magento2/Service/Shipping/LetterboxPackage.php on line 211 {"exception":"[object] (Exception(code: 0): Notice: Trying to access array offset on value of type null in /data/web/magento2/vendor/tig/postnl-magento2/Service/Shipping/LetterboxPackage.php on line 211 at /data/web/magento2/vendor/magento/framework/App/ErrorHandler.php:61)"} []
`

The code were this originates from:

```
public function fixConfigurableProductQty($products)
    {
        foreach($products as $product) {
            if ($product->getProductType() === 'configurable') {

                $product->getChildren()[0]->setQty($product->getQty());  // error happens here
            }
        }

        return $products;
    }

```

**To Reproduce**
1. Add configurable product to cart
2. Go to checkout & fill-in details
3. Try to place order

**Expected result**
Redirect to payment provider

**Actual result**
A server error emerges on the checkout page, thus preventing users to place & pay an order with configurable products

**Workaround**
None

**Screenshots**
![image](https://user-images.githubusercontent.com/57094396/140937395-7ba19427-7ae8-40b3-bb0d-dc4796774a6d.png)

- Magento version: 2.4.3-p1
- PHP version: 7.4
- TIG PostNL version: 1.10.0

**Comment by tig-jeffreybranderhorst** [comment]
Hi @rjm87 ,

Does this happen only with the payment provider options? Or also with the Magento payment options? And is the order you are trying to check out with a letterbox product? 

Have a great day,
Jeffrey

**Comment by rjm87** [comment]
Hi Jeffrey,

Thanks for the fast reply. I don't think it is a letterbox product, see the (default) settings of (one of) a configurable product which gives the error on the checkout page:

![image](https://user-images.githubusercontent.com/57094396/140940365-4a279ac8-cb3c-400b-b0ef-c7b50e097c62.png)

I disabled the Mollie_Payment plugin, and tried with the default Magento payment option, but the same error occurs:
![image](https://user-images.githubusercontent.com/57094396/140941196-df325547-3505-465d-ac1a-aa7ed736d4d3.png)

**Comment by tig-jeffreybranderhorst** [comment]
Hi @rjm87 ,

Thank you for the clarification!  I have tried to reproduce this on our test Magento version: 2.4.3-p1 with PostNL 1.10. But I'm not able to reproduce this issue, not with a Magento payment method or with an Ideal payment method.  Could you contact us at support@tig.nl? So we can have a quick look at your web shop and maybe see a difference between our test shop and yours.

Have a great day,
Jeffrey

**Comment by rjm87** [comment]
I've sent an email to support. FYI; Just downgraded to version 1.9.9 and the error is gone for now.

**Comment by tig-jeffreybranderhorst** [comment]
Hi @rjm87 ,

We have received the mail and will assist you further in the mail! 

Have a great day,
Jeffrey

**Comment by gjportegies** [comment]
We're also experiencing this issue on a Magento 2.4.3 instance with PostNL module version 1.10.0.

We've temporarily fixed the issue by surrounding line 211 with an issue. We're not using Letterbox Packages at all.

- We're also using Mollie Payments.

**Comment by tig-jeffreybranderhorst** [comment]
Hi @gjportegies ,

Thank you for submitting this, good to hear you already found a fix. We are investigating this, so we make a fix for this in the next version. 
Thank you for your info that help us with the investigation. 

Have a great day!
Jeffrey

**Comment by gjportegies** [comment]
Hi @tig-jeffreybranderhorst, just to clarify. Our solution is not a fix but just bypassing the issue since we're not using Letterbox Packages.

**Comment by tig-jeffreybranderhorst** [comment]
Hi @gjportegies ,

Correct, I worded it wrong. Thank you for your response!

Have a great day,
Jeffrey

**Comment by sumil3890** [comment]
@gjportegies Hello,
Can you please let me know which changes you applied exactly for workaround ?

**Comment by gjportegies** [comment]
@sumil3890 we've bypassed the issue by chaging the method to the following:
```
    public function fixConfigurableProductQty($products)
    {
        foreach($products as $product) {
            if ($product->getProductType() === 'configurable') {
                if (isset($product->getChildren()[0])) {
                        $product->getChildren()[0]->setQty($product->getQty());
                }
            }
        }

        return $products;
    }
```

**Comment by tig-jeffreybranderhorst** [comment]
Hi @rjm87 @gjportegies and @sumil3890, 

we have made a 1.10.1 release with a fix for this issue. Could test on a staging environment if this also solves your problem?

Have a great day,
Jeffrey

**Comment by gjportegies** [comment]
@tig-jeffreybranderhorst I've already tested this morning and it should work properly now.

**Comment by tig-jeffreybranderhorst** [comment]
Hi @gjportegies ,

Thank you for the update! That great to read! 

Have a great day,
Jeffrey

**Comment by rjm87** [comment]
@tig-jeffreybranderhorst I can confirm the issue is fixed in 1.10.1. Thanks guys!

**Comment by tig-jeffreybranderhorst** [comment]
Hi @rjm87,

Thank you for the update! That is great to read!

Have a great day,
Jeffrey
